### PR TITLE
fix: workload selector label exceeding 63-character limit

### PR DESCRIPTION
**Pull request by mbologna**
## Issue

When creating deployments via Rancher, the automatically generated label `workload.user.cattle.io/workloadselector` can exceed Kubernetes' 63-character limit for label values, causing deployment creation to fail.

## Problem

Rancher generates workload selector labels using the format `<type>-<namespace>-<deployment>` (e.g., `deployment-my-namespace-my-app`). This creates a hard limit where namespace + deployment name cannot exceed 46 characters combined, even though Kubernetes allows up to 63 characters for both namespace and deployment names individually.

Root cause: The label generation in `resolveWorkloadID()` didn't account for Kubernetes' 63-character label value limit, and there was no truncation or validation logic.

## Solution

Updated the workload selector label generation to use `SafeConcatName` from `pkg/capr/common.go`:

- Short names (≤63 chars): No change, uses original format `deployment-namespace-name`
- Long names (>63 chars): Truncates and appends hash, e.g., `deployment-very-long-namespace-very-long-dep-a1b2c`
- Stores full workload ID in annotation `workload.user.cattle.io/workload-id-full` when truncation occurs

### Manual Testing

1. **Short names (baseline)**: Created deployment with namespace `test-short` + name `nginx` → label: `deployment-test-short-nginx` 
2. **Boundary case (63 chars)**: Created deployment with 26-char namespace + 25-char name → label: full 63-char format 
3. **Over limit (64+ chars)**: Created deployment with 63-char namespace + 63-char name → label: truncated with hash, ≤63 chars 
4. **Annotation verification**: Long names correctly stored full ID in annotation 
5. **Pod selector**: Verified pods match deployment selector with hash-based labels 
6. **Determinism**: Same namespace+name produces identical hash across recreations 

Before fix: Deployments with namespace+deployment > 46 chars failed with "must be no more than 63 bytes"

After fix: Can create deployments with full 63-char namespace and deployment names

### Automated Testing

**Unit Tests**: Added comprehensive test suite in `aggregate_store_test.go`:
- `TestResolveWorkloadID_ShortNames`: Verifies backward compatibility
- `TestResolveWorkloadID_LongNames`: Verifies truncation + hash for long names
- `TestResolveWorkloadID_Determinism`: Verifies consistent hash generation
- `TestResolveWorkloadID_Uniqueness`: Verifies different inputs produce different hashes
- `TestResolveWorkloadID_BoundaryCase`: Tests exact 63/64 char boundary
- `TestResolveWorkloadID_DifferentTypes`: Tests all workload types

## Related PRs

Dashboard PR: https://github.com/rancher/dashboard/pull/18242 (must be deployed together)

### CODE REVIEW

File: pkg/api/norman/store/workload/aggregate_store.go [hunk 1]
```diff
@@ -142,7 +146,33 @@ func store(registries map[string]projectclient.RegistryCredential, domainToCreds
...
+		return workloadID
+	}
+
+	// For long names, store full ID in annotation for debugging
+	annotations := convert.ToMapInterface(data["workloadAnnotations"])
+	if annotations == nil {
+		annotations = make(map[string]interface{})
+	}
+	annotations[WorkloadIDAnnotation] = workloadID
+	data["workloadAnnotations"] = annotations
```
Comment (aruiz14) [inline pkg/api/norman/store/workload/aggregate_store.go]: I'm not really familiar with this code, but to me it feels odd that a method named "resolve" would actually modify the contents of `data`. Maybe do that in the caller instead, creating a separate function for the "abbreviated" name?
We could also argue if we should add a new annotation for *all* objects instead, for consistency.

File: pkg/api/norman/store/workload/aggregate_store.go [hunk 2]
```diff
@@ -142,7 +146,33 @@ func store(registries map[string]projectclient.RegistryCredential, domainToCreds
...
+
+	// Create hash-based identifier to stay within Kubernetes limits
+	// Format: deployment-<12-char-hash>
+	// This allows full 63-character namespace and deployment names
+	hasher := sha256.New()
```
Comment (manno) [inline pkg/api/norman/store/workload/aggregate_store.go]: Is it possible to reuse SafeConcatName? It produces human-readable output with a hash that is collision-resistant.

File: pkg/api/norman/store/workload/aggregate_store.go (hunk 1 above)
Comment (mbologna) [reply pkg/api/norman/store/workload/aggregate_store.go]: You're right. I now moved annotation logic to caller

File: pkg/api/norman/store/workload/aggregate_store.go (hunk 2 above)
Comment (mbologna) [reply pkg/api/norman/store/workload/aggregate_store.go]: OK I've reused it so there is consistency with other Rancher components
